pkg/plugins/k8s/k8s_plugin.go

```diff
@@ -0,0 +1,350 @@
+	if u.switchdevService {
+		updateList = append(updateList, "SwitchdevService")
+	}
+	if u.switchdevScript {
+		updateList = append(updateList, "SwitchdevScript")
+	}
+	for _, s := range u.systemServices {
+		updateList = append(updateList, s.Name)
+	}
+
+	return strings.Join(updateList, ",")
+}
+
+const (
+	switchdevManifestPath     = "bindata/manifests/switchdev-config/"
+	switchdevUnits            = switchdevManifestPath + "switchdev-units/"
+	switchdevUnitFile         = switchdevUnits + "switchdev-configuration.yaml"
+	networkManagerUnitFile    = switchdevUnits + "NetworkManager.service.yaml"
+	ovsUnitFile               = switchdevManifestPath + "ovs-units/ovs-vswitchd.service.yaml"
+	configuresSwitchdevScript = switchdevManifestPath + "files/configure-switchdev.sh.yaml"
+
+	chroot = "/host"
+)
+
+var (
+	Plugin K8sPlugin
+)
+
+// Initialize our plugin and set up initial values
+func init() {
+	Plugin = K8sPlugin{
+		PluginName:     "k8s_plugin",
+		SpecVersion:    "1.0",
+		serviceManager: service.NewServiceManager(chroot),
+		updateTarget:   &k8sUpdateTarget{},
+	}
+
+	// Read manifest files for plugin
+	if err := Plugin.readManifestFiles(); err != nil {
+		panic(err)
+	}
+}
+
+// Name returns the name of the plugin
+func (p *K8sPlugin) Name() string {
+	return p.PluginName
+}
+
+// Spec returns the version of the spec expected by the plugin
+func (p *K8sPlugin) Spec() string {
+	return p.SpecVersion
+}
+
+// OnNodeStateAdd Invoked when SriovNetworkNodeState CR is created, return if need dain and/or reboot node
+func (p *K8sPlugin) OnNodeStateAdd(state *sriovnetworkv1.SriovNetworkNodeState) (needDrain bool, needReboot bool, err error) {
+	glog.Info("k8s-plugin OnNodeStateAdd()")
+	return p.OnNodeStateChange(nil, state)
+}
+
+// OnNodeStateChange Invoked when SriovNetworkNodeState CR is updated, return if need dain and/or reboot node
+func (p *K8sPlugin) OnNodeStateChange(old, new *sriovnetworkv1.SriovNetworkNodeState) (needDrain bool, needReboot bool, err error) {
+	glog.Info("k8s-plugin OnNodeStateChange()")
+	needDrain = false
+	needReboot = false
+
+	p.updateTarget.reset()
+	// Check services
+	err = p.servicesStateUpdate()
+	if err != nil {
+		glog.Errorf("k8s-plugin OnNodeStateChange(): failed : %v", err)
+		return
+	}
+
+	// Check switchdev config
+	var update, remove bool
+	if update, remove, err = utils.WriteSwitchdevConfFile(new); err != nil {
+		glog.Errorf("k8s-plugin OnNodeStateChange():fail to update switchdev.conf file: %v", err)
+		return
+	}
+	if remove {
+		glog.Info("k8s-plugin OnNodeStateChange(): need reboot node to clean switchdev VFs")
+		needDrain = true
+		needReboot = true
+		return
+	}
+	if update {
+		glog.Info("k8s-plugin OnNodeStateChange(): need reboot node to use the up-to-date switchdev.conf")
+		needDrain = true
+		needReboot = true
+		return
+	}
+	if p.updateTarget.needUpdate() {
+		glog.Infof("k8s-plugin OnNodeStateChange(): needDrain to update %q", p.updateTarget)
+		needDrain = true
```

Comment:
don't you need reboot?

Reply:
Adding the services doesn't require reboot

Reply:
but what about the VF rep udev rename? it won't be triggered ...

Reply:
It is possible to add an additional condition to the udev renaming to do reboot, but the PR of renaming is not merged yet

Reply:
Sorry I don't follow why it better to do it in the udev PR and not here?

Reply:
I assume would  be better because the udev is not merged yet, and when udev is merged we need to do another PR to support it for k8s_plugin